src/controllers/handlers/community-voice-chat/community-voice-chat-handler.ts

```diff
@@ -0,0 +1,145 @@
+import { HandlerContextWithPath } from '../../../types'
+import { InvalidRequestError } from '../../../types/errors'
+import { CommunityVoiceChatAction } from '../../../types/community-voice'
+
+export async function communityVoiceChatHandler(
+  context: HandlerContextWithPath<'logs' | 'voice', '/community-voice-chat'>
+) {
+  const {
+    components: { logs, voice }
+  } = context
+  const logger = logs.getLogger('community-voice-chat-handler')
+
+  let body: {
+    community_id: string
+    user_address: string
+    action: CommunityVoiceChatAction
+  }
+
+  try {
+    body = await context.request.json()
+  } catch (error) {
+    logger.error(`Error parsing request body: ${error}`)
+    throw new InvalidRequestError('Invalid request body')
+  }
+
+  if (!body.community_id) {
+    throw new InvalidRequestError('The property community_id is required')
+  }
+
+  if (!body.user_address) {
+    throw new InvalidRequestError('The property user_address is required')
+  }
+
+  if (!body.action || !Object.values(CommunityVoiceChatAction).includes(body.action)) {
+    throw new InvalidRequestError(
+      'The property action is required and must be one of: create, join, request-to-speak, promote-speaker, demote-speaker, kick-player'
+    )
+  }
+
+  const lowerCaseUserAddress = body.user_address.toLowerCase()
+
+  try {
+    switch (body.action) {
+      case CommunityVoiceChatAction.CREATE: {
+        logger.debug('Creating community voice chat credentials for moderator')
+        const credentials = await voice.getCommunityVoiceChatCredentialsForModerator(
+          body.community_id,
+          lowerCaseUserAddress
+        )
+        logger.debug(
+          `Created community voice chat credentials for moderator ${lowerCaseUserAddress} in community ${body.community_id}`
+        )
+        return {
+          status: 200,
+          body: {
+            connection_url: credentials.connectionUrl
+          }
+        }
+      }
+
+      case CommunityVoiceChatAction.JOIN: {
+        logger.debug('Joining community voice chat as member')
+        const credentials = await voice.getCommunityVoiceChatCredentialsForMember(
+          body.community_id,
+          lowerCaseUserAddress
+        )
+        logger.info(
+          `Community voice chat access granted for member ${lowerCaseUserAddress} in community ${body.community_id}`
+        )
+        return {
+          status: 200,
+          body: {
+            connection_url: credentials.connectionUrl
+          }
+        }
+      }
+
+      case CommunityVoiceChatAction.REQUEST_TO_SPEAK: {
```

Comment:
This endpoint is correct, but if you want, there's an alternative to have an specific endpoint to request to speak in the voice chat:
`POST /community-voice-chat/:communityId/users/:address/attention` or some other wording that could represent the state.

Do we need an endpoint / action to remove the request to speak as well?